packages/graphql-server/src/makeMergedSchema/makeMergedSchema.ts

```diff
@@ -70,22 +70,44 @@ const mapFieldsToService = ({
  * If null or invalid value is returned, will trigger a GQL error
  */
 const resolveUnionType = (types: readonly GraphQLObjectType[]) => ({
-  __resolveType(obj: GraphQLObjectType) {
-    // resolves type of object by looking for the largest intersection of common fields
+  __resolveType(obj: Record<string, unknown>) {
+    // if obj has __typname, check that first to resolve type, otherwise, look for largest intersection
+    if (Object.prototype.hasOwnProperty.call(obj, '__typename')) {
```

Comment:
Only suggesting this cause I had to look stuff up, but MDN recommends we use `Object.hasOwn` instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn

<img width="757" alt="image" src="https://user-images.githubusercontent.com/32992335/178233415-341f8a79-a439-4d44-8f89-a952aabc97f5.png">

```suggestion
    if (Object.hasOwn(obj, '__typename')) {
```

But do we need to be this safe here? (I.e., we can't use `in`?) Better safe than sorry for sure, but I had a look at `obj.__proto__` in the example tal posted and it was `null`. But not sure if that was specific to that example or not. 🤔